sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -305,4 +329,127 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
     coeQuery.orderBy("disbursementSchedule.coeStatus");
     return coeQuery.getMany();
   }
+
+  /**
+   * Returns Disbursement and application details for COE Application details.
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application details.
+   */
+  async getDisbursementAndApplicationDetails(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "disbursementSchedule.coeStatus",
+        "disbursementSchedule.application",
+        "disbursementSchedule.coeDeniedOtherDesc",
+      ])
+      .innerJoinAndSelect("disbursementSchedule.application", "application")
+      .innerJoinAndSelect("application.location", "location")
+      .innerJoinAndSelect("application.student", "student")
+      .innerJoinAndSelect("student.user", "user")
+      .innerJoinAndSelect("application.offering", "offering")
+      .innerJoinAndSelect("offering.educationProgram", "educationProgram")
+      .leftJoinAndSelect(
+        "disbursementSchedule.coeDeniedReason",
+        "coeDeniedReason",
+      )
+      .where("location.id = :locationId", { locationId })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
+      })
+      .andWhere("disbursementSchedule.id = :disbursementScheduleId", {
+        disbursementScheduleId,
+      })
+      .getOne();
+  }
+
+  /**
+   * Summary of disbursement and application for Approval/Denial of COE.
+   * @param locationId
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application Summary.
+   */
+  async getDisbursementAndApplicationSummary(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "application.id",
+        "application.applicationStatus",
+        "application.assessmentWorkflowId",
+      ])
+      .innerJoin("disbursementSchedule.application", "application")
+      .innerJoin("application.location", "location")
+      .where("location.id = :locationId", { locationId })
+      .andWhere("disbursementSchedule.id = :disbursementScheduleId", {
+        disbursementScheduleId,
+      })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
```

Comment:
I see that this service is used in the `confirm` and `deny` controller and I believe that COE status cant be updated  in a `completed` application. do we need `ApplicationStatus.completed`?

Reply:
In my understanding @ann-aot, the first COE, once completed, will also mark the applications status as completed. So, when the second COE is completed the application status will be already in completed state.

Reply:
mmm, okay so about the application status for the scenario, where 1st COE is approved, so application status is now completed and for the same application if 2nd COE is denied, what will be application status?

Reply:
That is the right explanation @ann-aot. Please let me know if you still want more context.

Reply:
@ann-aot in that case where 2nd COE is declined, the application status will not change. To the best of my knowledge 2nd COE is to stop the money from being disbursed to student in unusual cases like student dropping out from college or something of that sort where their application is complete but something happens during the course of their study. 

@abschwenker Correct me if I am wrong.